File: app/src/main/java/hexlet/code/games/GreetGame.java
package hexlet.code.games;

import hexlet.code.GameEngine;

public class GreetGame extends Game {

    @Override
    public String getRule() {
        throw new IllegalCallerException("GreetGame can not start round");
    }

    @Override
    public GameRound getNewRound() {
        throw new IllegalCallerException("GreetGame can not start round");
    }

    @Override
    public void run(GameEngine runner) {
        // do nothing
    }
}

File: app/src/main/kotlin/hexlet/code/GreetGame.kt
package hexlet.code

class GreetGame : Game() {
    override fun rule(): String {
        throw NotImplementedError("GreetGame does not have rule")
    }

    override fun newRound(): GameRound {
        throw NotImplementedError("GreetGame can not start round")
    }

    override fun run(runner: GameRunner) {
        // do nothing
    }
}
